extension/js/common/settings.ts

```diff
@@ -276,8 +277,14 @@ export class Settings {
           window.location.href = Url.create('/chrome/settings/setup.htm', { acctEmail: response.acctEmail, idToken: response.id_token });
         }
       } else if (response.result === 'Denied' || response.result === 'Closed') {
-        if (settingsTabId) {
-          await Settings.renderSubPage(acctEmail, settingsTabId, '/chrome/settings/modules/auth_denied.htm');
+        const authDeniedHtml = await Api.ajax({ url: '/chrome/settings/modules/auth_denied.htm' }, Catch.stackTrace()) as string; // tslint:disable-line:no-direct-ajax
```

Comment:
Here, since it's just a bunch of text, I'd just make a private method like `getAuthDeniedHtml` that does something like
```ts
return [
  'whatever',
  'second line',
  ...
].join('<br>');
```
I think it will be "more together" then keeping it in a separate file.

Then you can delete auth_denied.html and auth_denied.ts

Reply:
If the two files are still used elsewhere, we should ideally refactor these situations too.

Reply:
You could put more stuff in the method to make it `renderAuthDeniedHtmlWithOptionToRetry` or whatever. The method we are in now is big enough already, anyway.

Reply:
![image](https://user-images.githubusercontent.com/2179726/96745222-904ba780-13ce-11eb-8583-7a038f0239a9.png)
it is used all across the app, that's why I didn't want to duplicate the text.

Reply:
Ah - didn't know. We can evaluate it in another PR. Ideally this would be just modals everywhere.